lib/manageiq/postgres_ha_admin/failover_monitor.rb

```diff
@@ -35,11 +35,11 @@ def monitor
           handler.do_before_failover
 
           new_conn_info = execute_failover(handler, server_store)
-          if new_conn_info
-            handler.do_after_failover(new_conn_info)
-          else
-            logger.error("#{log_prefix(__callee__)} Failover failed")
-          end
+
+          logger.error("#{log_prefix(__callee__)} Failover failed") unless new_conn_info
+
+          # Upon success, we pass a connection hash, otherwise false
+          handler.do_after_failover(new_conn_info)
```

Comment:
As discussed, let's leave this code as is.  It's weird to call `after_failover` when you haven't actually failed over. I could see adding a new hook in the old else condition called `failed_failover` or something, but since we don't have a use case on the caller, we can do that separately.

If we keep this hook code as is, then the only changes need on the server side are to remove the LinuxAdmin::Service stop and restart calls (which aren't need when systemd is doing all the work along with manageiq-db-ready.